### PR TITLE
Fully switch master branch to MuseScore 4

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
for now mainly for the builds not to collide with the 3.x ones, as far as their settings are concerned.

To do: update the two splash screen PNGs

### CODE REVIEW

File: mscore/musescore.cpp [hunk 1]
```diff
@@ -7390,12 +7390,12 @@ MuseScoreApplication* MuseScoreApplication::initApplication(int& argc, char** ar
...
-            appName2 = "mscore-dev3";
-            appName  = "MuseScore3Development";
+            appName2 = "mscore-dev4";
+            appName  = "MuseScore4Development";
             }
       else {
-            appName2 = "mscore3";
-            appName  = "MuseScore3";
+            appName2 = "mscore4";
+            appName  = "MuseScore4";
```
Comment (Howard-C) [inline mscore/musescore.cpp:7398]: While we're at them, isn't it better to put spaces between words? `MuseScore 4 Dev` / `MuseScore 4`? Are there downsides for using spaces in this particular scenario?

Comment (Jojo-Schmitz) [reply mscore/musescore.cpp:7398]: I'd rather not do that.

Comment (Howard-C) [reply mscore/musescore.cpp:7398]: But they're also used for some dialogues in the app, and it's not consistent to see "MuseScore 4" on the main window but "MuseScore4" for others.

Comment (Jojo-Schmitz) [reply mscore/musescore.cpp:7398]: Where exactly? Anyhow, I see no compelling reason to change this.

Comment (Jojo-Schmitz) [reply mscore/musescore.cpp:7398]: As far as I can tell that `appName` is only used for path names (and for telemetry)

**Comment by Jojo-Schmitz** [comment]
Just found this:

    #if !defined(Q_OS_MAC) && (!defined(Q_OS_WIN) || defined(MSCORE_UNSTABLE))
    //---------------------------------------------------------
    //   default period
    //---------------------------------------------------------
    
    static int defaultPeriod()
          {
          int result = 24;
          if(qApp->applicationName() == "MuseScore2"){ //avoid nightly cymbals
                if (MuseScore::unstable())
                      result = 24;
                else
                      result = 24; // yes, it's again the same but let's keep the logic for now
                }
          return result;
          }
    #endif


Would that need to get adjusted too? What's the purpose actually?

**Comment by anatoly-os** [comment]
This is for the builds that don't use Sparkle. But the comparison with MuseScore 2 looks weird... We probably forgot to remove it before MuseScore 3 release.

**Comment by Jojo-Schmitz** [comment]
want another PR to fix this?

**Pull request merged**